torch/nativert/graph/GraphSignature.cpp

```diff
@@ -0,0 +1,479 @@
+      argName = inputSpec.get_custom_obj().get_arg().get_name();
+      break;
+    case torch::_export::InputSpec::Tag::USER_INPUT:
+      tagName = "USER_INPUT";
+      if (inputSpec.get_user_input().get_arg().tag() ==
+          torch::_export::Argument::Tag::AS_TENSOR) {
+        argName =
+            inputSpec.get_user_input().get_arg().get_as_tensor().get_name();
+      } else if (
+          inputSpec.get_user_input().get_arg().tag() ==
+          torch::_export::Argument::Tag::AS_CUSTOM_OBJ) {
+        argName =
+            inputSpec.get_user_input().get_arg().get_as_custom_obj().get_name();
+      } else {
+        TORCH_CHECK(false, "Unsupported USER_INPUT argument type.");
+      }
+      break;
+    case torch::_export::InputSpec::Tag::CONSTANT_INPUT:
+      argName = inputSpec.get_constant_input().get_name();
+      tagName = "CONSTANT_INPUT";
+      break;
+    case torch::_export::InputSpec::Tag::TOKEN:
+      TORCH_CHECK(false, "Token inputs not implemented yet.");
+    default:
+      TORCH_CHECK(false, "Unknown InputSpec tag encountered.");
+  }
+  return std::make_pair(std::move(argName), std::move(tagName));
+}
+
+void checkInputOrders(
+    const std::vector<torch::_export::InputSpec>& inputSpecs) {
+  // Map each tag to its index in the expected order
+  static constexpr std::
+      array<std::pair<torch::_export::InputSpec::Tag, uint32_t>, 5>
+          tagOrderArray = {
+              {{torch::_export::InputSpec::Tag::TOKEN, 0},
+               {torch::_export::InputSpec::Tag::PARAMETER, 1},
+               {torch::_export::InputSpec::Tag::BUFFER, 2},
+               {torch::_export::InputSpec::Tag::TENSOR_CONSTANT, 3},
+               {torch::_export::InputSpec::Tag::CUSTOM_OBJ, 4}}};
+  uint32_t currentOrderIndex = 0;
+  bool seenNonPersistentBuffer = false;
+  for (const auto& inputSpec : inputSpecs) {
+    if (inputSpec.tag() == torch::_export::InputSpec::Tag::USER_INPUT ||
+        inputSpec.tag() == torch::_export::InputSpec::Tag::CONSTANT_INPUT) {
+      continue;
+    }
+    auto it = std::find_if(
+        tagOrderArray.begin(),
+        tagOrderArray.end(),
+        [&inputSpec](const auto& pair) {
+          return pair.first == inputSpec.tag();
+        });
+    TORCH_CHECK(
+        it != tagOrderArray.end(), "Unknown InputSpec tag encountered.");
+    uint32_t tagIndex = it->second;
+    if (tagIndex < currentOrderIndex) {
+      auto [argName, tagName] = getSpecDetails(inputSpec);
+      TORCH_CHECK(
+          false,
+          fmt::format(
+              "Input arg {} with InputSpec {} is out of order!",
+              argName,
+              tagName));
+    }
+    currentOrderIndex = tagIndex;
+    // Additional check for buffers
+    if (inputSpec.tag() == torch::_export::InputSpec::Tag::BUFFER) {
+      if (!inputSpec.get_buffer().get_persistent()) {
+        seenNonPersistentBuffer = true;
+      } else if (
+          inputSpec.get_buffer().get_persistent() && seenNonPersistentBuffer) {
```

Comment:
why do you need to repeat the check of `inputSpec.get_buffer().get_persistent()`? couldn't you just write ```
else {
    TORCH_CHECK(!seenNonPersistentBuffer, "Persistent buffers must come before non-persistent buffers.");
}
```